packages/desktop-client/src/components/reports/ReportOptions.tsx

```diff
@@ -44,3 +51,132 @@ const intervalOptions = [
 { value: 5, description: 'Yearly', name: 5,
 ];
 */
+export type QueryDataEntity = {
+  date: string;
+  category: string;
+  categoryGroup: string;
+  account: string;
+  accountOffBudget: boolean;
+  payee: string;
+  transferAccount: string;
+  amount: number;
+};
+
+export type UncategorizedEntity = {
+  id?: string;
+  name: string;
+  hidden?: boolean;
+  uncat_id: string;
+  offBudget: boolean;
+  transfer: boolean;
+  category: boolean;
+};
+
+export type UncategorizedGroupEntity = {
+  id?: string;
+  name: string;
+  hidden?: boolean;
+  categories?: UncategorizedEntity[];
+};
+
+let uncategorizedCategory: UncategorizedEntity = {
+  name: 'Uncategorized',
+  id: null,
+  uncat_id: '1',
+  hidden: false,
+  offBudget: false,
+  transfer: false,
+  category: false,
+};
+let transferCategory: UncategorizedEntity = {
+  name: 'Transfers',
+  id: null,
+  uncat_id: '2',
+  hidden: false,
+  transfer: true,
+  offBudget: false,
+  category: false,
+};
+let offBudgetCategory: UncategorizedEntity = {
+  name: 'Off Budget',
+  id: null,
+  uncat_id: '3',
+  hidden: false,
+  offBudget: true,
+  transfer: false,
+  category: true,
+};
+
+let uncategouncatGrouprizedGroup: UncategorizedGroupEntity = {
+  name: 'Uncategorized & Off Budget',
+  id: null,
+  hidden: false,
+  categories: [uncategorizedCategory, transferCategory, offBudgetCategory],
+};
+
+type categoryListsProps = {
+  hidden: boolean;
+  uncat: boolean;
+  categories: { list: CategoryEntity[]; grouped: CategoryGroupEntity[] };
+};
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏these are not really `props`. Component have `props`. Utility functions have `arguments/parameters`. 

Also: this is a total nitpick, but I'd recommend just inlining the type definition for utility functions. Makes your life easier as you won't need to create a named type.